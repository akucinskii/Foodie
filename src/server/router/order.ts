import { createRouter } from "./context";
import { z } from "zod";
import { McListItemType, McListType } from "../../pages/Client/[orderId]";
import { OrderSlice } from "@prisma/client";
import dayjs from "dayjs";

export const orderRouter = createRouter()
  .query("hello", {
    input: z
      .object({
        text: z.string().nullish(),
      })
      .nullish(),
    resolve({ input }) {
      return {
        greeting: `Hello ${input?.text ?? "world"}`,
      };
    },
  })
  .query("getAllOrders", {
    async resolve({ ctx }) {
      return await ctx.prisma.order.findMany();
    },
  })
  .query("getAllTodayOrders", {
    async resolve({ ctx }) {
      let lastDay: string | number = Date.now() - 24 * 60 * 60 * 1000;
      lastDay = new Date(lastDay).toISOString();

      const response = await ctx.prisma.order.findMany({
        where: {
          createdAt: {
            gte: lastDay,
          },
        },
      });
      return response;
    },
  })
  .query("getAllOrderSlices", {
    async resolve({ ctx }) {
      return await ctx.prisma.orderSlice.findMany();
    },
  })
  .query("getOrderSlicesByOrderId", {
    input: z.object({
      id: z.string(),
    }),
    async resolve({ ctx, input }) {
      return await ctx.prisma.orderSlice.findMany({
        where: {
          orderId: input.id,
        },
      });
    },
  })
  .query("getOrderDetails", {
    input: z.object({
      id: z.string(),
    }),
    async resolve({ ctx, input }) {
      const response = await ctx.prisma.order.findUnique({
        where: {
          id: input.id,
        },
        include: {
          orderSlices: true,
        },
      });

      if (response) {
        const notMergedDetails: McListType[] = response.orderSlices.map(
          (orderSlice: OrderSlice) => {
            const parsed = JSON.parse(orderSlice.details) as McListType;
            return parsed;
          }
        );

        if (notMergedDetails.length === 1) {
          return notMergedDetails[0];
        }

        const mergedArray: McListItemType[] = [];

        if (notMergedDetails[0]) {
          const mergedDetails = notMergedDetails[0].concat(...notMergedDetails);

          mergedDetails.forEach((item) => {
            const found = mergedArray.find(
              (helperItem) => helperItem.id === item.id
            );
            if (found) {
              found.quantity += item.quantity;
            } else {
              mergedArray.push(item);
            }
          });

          return mergedArray;
        }
      }
    },
  })
  .query("getOrderSlicesAuthors", {
    input: z.object({
      id: z.string(),
    }),
    async resolve({ ctx, input }) {
      const response = await ctx.prisma.order.findUnique({
        where: {
          id: input.id,
        },
        include: {
          orderSlices: true,
        },
      });

      if (response) {
        const authors: { [key: string]: number } = {};
        response.orderSlices.forEach((orderSlice) => {
          const OrderSliceAccumulatedPrice = JSON.parse(
            orderSlice.details
          ).reduce(
            (acc: number, curr: McListItemType) =>
              acc + curr.quantity * curr.price,
            0
          );
          if (authors[orderSlice.author]) {
            authors[orderSlice.author] += OrderSliceAccumulatedPrice;
          } else {
            authors[orderSlice.author] = OrderSliceAccumulatedPrice;
          }
        });

        return authors;
      }
    },
  })
  .mutation("createOrderSlice", {
    input: z.object({
      details: z.string(),
      orderId: z.string(),
      author: z.string(),
    }),
    async resolve({ ctx, input }) {
      return await ctx.prisma.orderSlice.create({
        data: {
          details: input.details,
          orderId: input.orderId,
          author: input.author,
        },
      });
    },
  })
  .mutation("updateOrderSlice", {
    input: z.object({
      id: z.string(),
      details: z.string(),
    }),
    async resolve({ ctx, input }) {
      return await ctx.prisma.orderSlice.update({
        where: {
          id: input.id,
        },
        data: {
          details: input.details,
        },
      });
    },
  })
  .mutation("deleteOrderSlice", {
    input: z.object({
      id: z.string(),
    }),
    async resolve({ ctx, input }) {
      return await ctx.prisma.orderSlice.delete({
        where: {
          id: input.id,
        },
      });
    },
  })
  .mutation("createOrder", {
    input: z.object({
      name: z.string(),
      author: z.string(),
    }),
    async resolve({ ctx, input }) {
      const currentDate = new Date();

      return await ctx.prisma.order.create({
        data: {
          author: input.author,
          name: input.name,
          createdAt: currentDate,
        },
      });
    },
  })
  .mutation("removeOrder", {
    input: z.object({
      id: z.string(),
    }),
    async resolve({ ctx, input }) {
      return await ctx.prisma.order.delete({
        where: {
          id: input.id,
        },
      });
    },
  });
