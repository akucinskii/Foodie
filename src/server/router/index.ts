// src/server/router/index.ts
import { createRouter } from "./context";
import superjson from "superjson";

import { orderRouter } from "./order";

export const appRouter = createRouter()
  .transformer(superjson)
  .merge("order.", orderRouter);

// export type definition of API
export type AppRouter = typeof appRouter;
