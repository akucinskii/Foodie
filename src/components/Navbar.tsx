import { signIn, signOut, useSession } from "next-auth/react";
import Image from "next/image";
import Link from "next/link";
import { useRouter } from "next/router";
import { router } from "../server/trpc/trpc";
import Button from "./Button";

const Navbar = () => {
  const { data: session } = useSession();
  const router = useRouter();

  return (
    <div className="align-center fixed z-20 flex w-full flex-row bg-white">
      <div className="flex-1">
        <ul className="menu menu-horizontal gap-4 p-2">
          <h1 className="text-4xl font-bold text-gray-700">
            Solve<span className="text-yellow-500">M</span>c
          </h1>

          <li>
            <Link href="/">
              <a className="whitespace-nowrap">Home</a>
            </Link>
          </li>

          <li>
            <Link href="/all">
              <a className="whitespace-nowrap">All Orders</a>
            </Link>
          </li>

          <li className="">
            <Link href="/Driver/newOrder">
              <a className="=whitespace-nowrap">New Order</a>
            </Link>
          </li>
        </ul>
      </div>
      <div className="flex-0">
        <ul className="align-center menu menu-horizontal gap-2 p-2">
          {session ? (
            <>
              <button
                onClick={() => router.push(`/Profile/${session.user?.id}`)}
                className="btn btn-ghost btn-circle"
              >
                {session && session.user && session.user.image && (
                  <Image
                    className="rounded-full"
                    src={session?.user?.image}
                    alt="text"
                    width={40}
                    height={40}
                  />
                )}
              </button>
              <li>
                <Button
                  onClick={() => {
                    router.push("/");
                    signOut();
                  }}
                >
                  Sign out
                </Button>
              </li>
            </>
          ) : (
            <Button
              onClick={() => {
                signIn();
              }}
            >
              Sign in
            </Button>
          )}
        </ul>
      </div>
    </div>
  );
};

export default Navbar;
