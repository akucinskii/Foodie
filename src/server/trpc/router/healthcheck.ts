import { publicProcedure, router } from "../trpc";

export const healthcheckRouter = router({
  /**
   * Healthcheck route
   *
   * @returns {Object} message
   *
   * @example
   * ```ts
   * const response = await trpc.healthcheck.healthcheck();
   * ```
   */
  healthcheck: publicProcedure.query(() => {
    return {
      message: "ok",
    };
  }),
});
