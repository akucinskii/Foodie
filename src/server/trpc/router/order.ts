import { z } from "zod";
import { router, protectedProcedure, publicProcedure } from "../trpc";

export const orderRouter = router({
  /**
   * Healthcheck. If this fails, everything else will also fail.
   *
   * @returns {string} "ok"
   *
   * @example
   * ```ts
   * const response = await trpc.order.healthcheck();
   * ```
   */
  healthcheck: publicProcedure.query(() => {
    return {
      message: "ok",
    };
  }),

  /**
   * Get all orders
   *
   * @returns order object Array
   *
   * @example
   * ```ts
   * const orders = await trpc.order.getAllOrders();
   * ```
   */
  getAllOrders: protectedProcedure.query(async ({ ctx }) => {
    return await ctx.prisma.order.findMany();
  }),

  /**
   * Get new date then set the hour to 00:00:00,
   * Then return all orders from that date
   *
   * @example
   * ```ts
   * const orders = await trpc.order.getOrdersByDate();
   * ```
   */
  getAllTodayOrders: publicProcedure.query(async ({ ctx }) => {
    const dateAsNumber = new Date().setHours(0, 0, 0, 0);
    const beginningOfToday = new Date(dateAsNumber);

    const response = await ctx.prisma.order.findMany({
      where: {
        createdAt: {
          gte: beginningOfToday,
        },
      },
    });

    return response;
  }),

  getOrderById: protectedProcedure
    .input(
      z.object({
        id: z.string(),
      })
    )
    .query(async ({ ctx, input }) => {
      const response = await ctx.prisma.order.findUnique({
        where: {
          id: input.id,
        },
        include: {
          orderSlices: {
            include: {
              author: true,
              OrderItem: {
                include: {
                  RestaurantMenuItem: true,
                },
                orderBy: {
                  quantity: "desc",
                },
              },
            },
          },
          Restaurant: {
            include: {
              RestaurantMenuItem: true,
            },
          },
        },
      });

      return response;
    }),

  /**
   * Creates a new order
   * Author should be taken from next-auth session user
   *
   * @param  {string} name - order name
   * @param {string} author - order author
   *
   * @example
   * ```ts
   * const response = await trpc.order.createOrder({
   *  name: "someName",
   * author: "someAuthor",
   * })
   */
  createOrder: protectedProcedure
    .input(
      z.object({
        name: z.string().max(50),
        author: z.string(),
        restaurantId: z.string(),
      })
    )
    .mutation(async ({ ctx, input }) => {
      const currentDate = new Date();
      return await ctx.prisma.order.create({
        data: {
          restaurantId: input.restaurantId,
          name: input.name,
          authorId: input.author,
          createdAt: currentDate,
        },
      });
    }),

  /**
   * Deletes order by provided id
   *
   * @param {string} id - order id
   *
   * @example
   * ```ts
   * const mutation = trpc.order.removeOrder.useMutation()
   *
   * mutation.mutateAsync({
   *    id: "1",
   * });
   * ```
   *
   */
  removeOrder: publicProcedure
    .input(
      z.object({
        id: z.string(),
      })
    )
    .mutation(async ({ ctx, input }) => {
      return await ctx.prisma.order.delete({
        where: {
          id: input.id,
        },
      });
    }),
});
