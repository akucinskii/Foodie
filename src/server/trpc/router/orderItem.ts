import { z } from "zod";
import { protectedProcedure, router } from "../trpc";

export const orderItemRouter = router({
  createOrderItem: protectedProcedure
    .input(
      z.object({
        quantity: z.number().min(1).max(50),
        restaurantMenuItemId: z.string(),
        orderSliceId: z.string(),
      })
    )
    .mutation(async ({ ctx, input }) => {
      const response = await ctx.prisma.orderItem.create({
        data: {
          quantity: input.quantity,
          restaurantMenuItemId: input.restaurantMenuItemId,
          orderSliceId: input.orderSliceId,
        },
      });
      return response;
    }),

  updateOrderItem: protectedProcedure
    .input(
      z.object({
        id: z.string(),
        quantity: z.number().min(1).max(50),
      })
    )
    .mutation(async ({ ctx, input }) => {
      const response = await ctx.prisma.orderItem.update({
        where: {
          id: input.id,
        },
        data: {
          quantity: input.quantity,
        },
      });
      return response;
    }),

  removeOrderItem: protectedProcedure
    .input(
      z.object({
        id: z.string(),
      })
    )
    .mutation(async ({ ctx, input }) => {
      const response = await ctx.prisma.orderItem.delete({
        where: {
          id: input.id,
        },
      });
      return response;
    }),
});
