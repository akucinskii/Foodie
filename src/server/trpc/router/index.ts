// src/server/router/index.ts

import { orderRouter } from "./order";
import { authRouter } from "./auth";
import { router } from "../trpc";
import { userRouter } from "./user";
import { healthcheckRouter } from "./healthcheck";
import { orderSliceRouter } from "./orderSlice";
import { randomNumberRouter } from "./randomNumber";
import { restaurantRouter } from "./restaurant";
import { restaurantMenuItemRouter } from "./restaurantMenuItem";
import { orderItemRouter } from "./orderItem";

export const appRouter = router({
  auth: authRouter,
  order: orderRouter,
  user: userRouter,
  orderSlice: orderSliceRouter,
  orderItem: orderItemRouter,
  healthcheck: healthcheckRouter,
  randomNumber: randomNumberRouter,
  restaurant: restaurantRouter,
  restaurantMenuItem: restaurantMenuItemRouter,
});
// export type definition of API
export type AppRouter = typeof appRouter;
