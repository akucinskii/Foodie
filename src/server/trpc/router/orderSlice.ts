import { protectedProcedure, router } from "../trpc";
import { z } from "zod";

export const orderSliceRouter = router({
  getAllOrderSlices: protectedProcedure.query(async ({ ctx }) => {
    return await ctx.prisma.orderSlice.findMany();
  }),

  /**
   * Get a single order slice by id
   *
   * @param id The id of the order slice to get
   *
   * @returns The order slice
   *
   * @throws 404 if the order slice is not found
   *
   * @example
   * ```ts
   * const orderSlice = await trpc.orderSlice.getOrderSlice({ id: "1" });
   * ```
   */
  getOrderSliceById: protectedProcedure
    .input(
      z.object({
        id: z.string(),
      })
    )
    .query(async ({ ctx, input }) => {
      const response = await ctx.prisma.orderSlice.findUnique({
        where: {
          id: input.id,
        },
      });

      if (!response) {
        throw new Error("OrderSlice not found");
      }

      const formattedDetails = {
        ...response,
        details: JSON.parse(response.details || "{}"),
      };
      return formattedDetails;
    }),

  /**
   * Return all order slices from a specific order
   *
   * @param orderId
   *
   * @returns OrderSlice[]
   *
   * @example
   * ```ts
   * const orderSlices = await trpc.orderSlice.getOrderSlicesByOrderId({
   *  orderId: "someOrderId",
   * })
   *
   * ```
   */
  getOrderSlicesByOrderId: protectedProcedure
    .input(
      z.object({
        id: z.string(),
      })
    )
    .query(async ({ ctx, input }) => {
      const response = await ctx.prisma.orderSlice.findMany({
        where: {
          orderId: input.id,
        },
        include: {
          author: true,
        },
      });

      const formattedDetails = response.map(
        (item: typeof response[number]) => ({
          ...item,
          details: JSON.parse(item.details),
        })
      );

      return formattedDetails;
    }),

  /**
   * Creates new orderSlice
   *
   * details must be  provided as string of json object.
   *
   * @param {string} orderId - id of order (the bigger one)
   * @param {string} details - string of json object
   * @param {string} author - id of user creating orderSlice
   *
   * @example
   * ```ts
   * const orderSlice = {
   *    "id": "1",
   *    "name": "McChicken",
   *    "price": 5,
   *    "quantity": 1
   * }
   * const orderSliceAsJson = JSON.stringify(orderSlice);
   * const mutation = trpc.order.createOrderSlice.useMutation()
   *
   * mutation.mutateAsync({
   *    orderId: "1",
   *    details: orderSliceAsJson,
   *    author: "Example author",
   * });
   * ```
   */
  createOrderSlice: protectedProcedure
    .input(
      z.object({
        details: z.string(),
        orderId: z.string(),
        author: z.string(),
      })
    )
    .mutation(async ({ ctx, input }) => {
      return await ctx.prisma.orderSlice.create({
        data: {
          details: input.details,
          orderId: input.orderId,
          authorId: input.author,
        },
      });
    }),

  /**
   * Updates orderSlice with details provided as string of json object.
   * @returns {Promise<Prisma.OrderSlice>}OrderSlice
   * @param {string} id
   * @param {string} data
   *
   *
   * @example
   * ```ts
   *
   * const mutation = trpc.order.updateOrderSlice.useMutation()
   *
   * mutation.mutateAsync({
   *    orderId: "1",
   *    details: orderSliceAsJson,
   * });
   * ```
   *
   *
   */
  updateOrderSlice: protectedProcedure
    .input(
      z.object({
        id: z.string(),
        details: z.string(),
      })
    )
    .mutation(async ({ ctx, input }) => {
      return await ctx.prisma.orderSlice.update({
        where: {
          id: input.id,
        },
        data: {
          details: input.details,
        },
      });
    }),

  /**
   * Deletes orderSlice
   * @param {string} id
   *
   * @example
   * ```ts
   * const mutation = trpc.order.deleteOrderSlice.useMutation()
   *
   * mutation.mutateAsync({
   *  id: "1",
   * })
   * ```
   */
  deleteOrderSlice: protectedProcedure
    .input(
      z.object({
        id: z.string(),
      })
    )
    .mutation(async ({ ctx, input }) => {
      return await ctx.prisma.orderSlice.delete({
        where: {
          id: input.id,
        },
      });
    }),
});
