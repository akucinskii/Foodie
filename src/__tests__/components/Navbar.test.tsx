import { fireEvent, render } from "@testing-library/react";
import { Session } from "next-auth";
import { describe, expect, test, vi, Mock, afterAll } from "vitest";
import { signIn, signOut, useSession } from "next-auth/react";
import Navbar from "../../components/Navbar";
import { useRouter } from "next/router";

const mockSession: Session = {
  user: {
    id: "1",
    name: "Test",
    email: "test@test.test",
    image: "https://testimages.org/img/testimages_screenshot.jpg",
  },
  expires: "1",
};

vi.mock("next-auth/react", () => ({
  signIn: vi.fn(),
  signOut: vi.fn(),
  useSession: vi.fn(() => ({
    data: mockSession,
    status: "unauthenticated",
  })),
}));

vi.mock("next/router", () => ({
  useRouter: vi.fn(() => ({
    router: {
      push: vi.fn(),
    },
  })),
}));

vi.mock("next-auth/react");

describe("Navbar", () => {
  (signIn as Mock).mockImplementation(() => vi.fn());
  (signOut as Mock).mockImplementation(() => vi.fn());

  const { container, rerender, getByText, getByTestId } = render(<Navbar />);

  test("should match snapshot when logout", () => {
    expect(container).toMatchSnapshot("logout");
  });

  test("should match snapshot when login", () => {
    (useSession as Mock).mockReturnValue({
      data: mockSession,
      status: "authenticated",
    });

    expect(container).toMatchSnapshot("login");
  });

  test("should call signIn when click on login button", async () => {
    (useSession as Mock).mockImplementation(() => ({
      data: null,
      status: "unauthenticated",
    }));

    rerender(<Navbar />);

    fireEvent.click(getByText("Sign in"));

    expect(signIn).toHaveBeenCalled();
  });

  test("should call signOut when click on logout button", async () => {
    const mockRouter = {
      push: vi.fn(), // the component uses `router.push` only
    };
    (useRouter as Mock).mockReturnValue(mockRouter);

    (useSession as Mock).mockReturnValue({
      data: mockSession,
      status: "authenticated",
    });

    rerender(<Navbar />);

    fireEvent.click(getByText("Sign out"));

    expect(signOut).toHaveBeenCalled();
    expect(mockRouter.push).toHaveBeenCalledWith("/");
  });
  test("should call router.push when click on profile button", async () => {
    const mockRouter = {
      push: vi.fn(), // the component uses `router.push` only
    };
    (useRouter as Mock).mockReturnValue(mockRouter);

    (useSession as Mock).mockReturnValue({
      data: mockSession,
      status: "authenticated",
    });

    rerender(<Navbar />);

    fireEvent.click(getByTestId("profile-image"));

    expect(mockRouter.push).toHaveBeenCalledWith(
      "/Profile/" + mockSession.user?.id
    );
  });

  afterAll(() => {
    (signIn as Mock).mockRestore();
    (signOut as Mock).mockRestore();
    (useSession as Mock).mockRestore();
  });
});
