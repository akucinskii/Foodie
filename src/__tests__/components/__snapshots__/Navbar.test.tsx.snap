// Vitest Snapshot v1

exports[`Navbar > should match snapshot when login > login 1`] = `
<div>
  <div
    class="align-center fixed z-20 flex w-full flex-row bg-white"
  >
    <div
      class="flex-1"
    >
      <ul
        class="menu menu-horizontal gap-4 p-2"
      >
        <h1
          class="text-4xl font-bold text-gray-700"
        >
          Solve
          <span
            class="text-yellow-500"
          >
            M
          </span>
          c
        </h1>
        <li>
          <a
            class="whitespace-nowrap"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="whitespace-nowrap"
            href="/all"
          >
            All Orders
          </a>
        </li>
        <li
          class=""
        >
          <a
            class="=whitespace-nowrap"
            href="/Driver/newOrder"
          >
            New Order
          </a>
        </li>
        <li
          class=""
        >
          <a
            class="=whitespace-nowrap"
            href="/random"
          >
            Randomize
          </a>
        </li>
      </ul>
    </div>
    <div
      class="flex-0"
    >
      <ul
        class="align-center menu menu-horizontal gap-2 p-2"
      >
        <button
          class="btn btn-ghost btn-circle"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="text"
              class="rounded-full"
              data-nimg="intrinsic"
              data-testid="profile-image"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </button>
        <li>
          <button
            class="btn btn-primary text-white"
            data-testid="button"
            type="button"
          >
            Sign out
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Navbar > should match snapshot when logout > logout 1`] = `
<div>
  <div
    class="align-center fixed z-20 flex w-full flex-row bg-white"
  >
    <div
      class="flex-1"
    >
      <ul
        class="menu menu-horizontal gap-4 p-2"
      >
        <h1
          class="text-4xl font-bold text-gray-700"
        >
          Solve
          <span
            class="text-yellow-500"
          >
            M
          </span>
          c
        </h1>
        <li>
          <a
            class="whitespace-nowrap"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="whitespace-nowrap"
            href="/all"
          >
            All Orders
          </a>
        </li>
        <li
          class=""
        >
          <a
            class="=whitespace-nowrap"
            href="/Driver/newOrder"
          >
            New Order
          </a>
        </li>
        <li
          class=""
        >
          <a
            class="=whitespace-nowrap"
            href="/random"
          >
            Randomize
          </a>
        </li>
      </ul>
    </div>
    <div
      class="flex-0"
    >
      <ul
        class="align-center menu menu-horizontal gap-2 p-2"
      >
        <button
          class="btn btn-ghost btn-circle"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="text"
              class="rounded-full"
              data-nimg="intrinsic"
              data-testid="profile-image"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </button>
        <li>
          <button
            class="btn btn-primary text-white"
            data-testid="button"
            type="button"
          >
            Sign out
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
